lib/matplotlib/dates.py

```diff
@@ -1113,7 +1208,8 @@ def __init__(self,  byhour=None, interval=1, tz=None):
         example, if ``interval=2``, mark every second occurrence.
         """
         if byhour is None:
-            byhour = list(xrange(24))
+            byhour = range(24)
```

Comment:
I can double check, but the API docs in dateutil specify that any sequence is fine, so if it does cause a problem, it's a problem with dateutil. I'm fairly certain that these `list(xrange(#))` lines are an artifact of 2to3, and not a deliberate choice that was made. 

There was a bug in dateutil <= 2.3 which had a problem with numpy arrays, but I've fixed that. In the unlikely event that range objects don't work with dateutil and Python 3, I'll fix that in dateutil and we can add a TODO to switch these back to `range()` objects if the dateutil dependency is ever raised to >= 2.5.0.